mod-audit-server/src/main/java/org/folio/verticle/acquisition/consumers/OrderEventsHandler.java

```diff
@@ -14,50 +12,38 @@
 import org.folio.rest.jaxrs.model.OrderAuditEvent;
 import org.folio.rest.util.OkapiConnectionParams;
 import org.folio.services.acquisition.OrderAuditEventsService;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
-import java.util.List;
-
 @Component
 public class OrderEventsHandler implements AsyncRecordHandler<String, String> {
 
   private static final Logger LOGGER = LogManager.getLogger();
 
-  private OrderAuditEventsService orderAuditEventsService;
-
-  private Vertx vertx;
+  private final OrderAuditEventsService orderAuditEventsService;
+  private final Vertx vertx;
 
-  public OrderEventsHandler(@Autowired Vertx vertx,
-                            @Autowired OrderAuditEventsService orderAuditEventsService) {
+  public OrderEventsHandler(Vertx vertx,
+                            OrderAuditEventsService orderAuditEventsService) {
     this.vertx = vertx;
     this.orderAuditEventsService = orderAuditEventsService;
   }
 
   @Override
-  public Future<String> handle(KafkaConsumerRecord<String, String> record) {
-
-    Promise<String> result = Promise.promise();
-    List<KafkaHeader> kafkaHeaders = record.headers();
-    OkapiConnectionParams okapiConnectionParams = new OkapiConnectionParams(KafkaHeaderUtils.kafkaHeadersToMap(kafkaHeaders), vertx);
-    OrderAuditEvent event = new JsonObject(record.value()).mapTo(OrderAuditEvent.class);
+  public Future<String> handle(KafkaConsumerRecord<String, String> kafkaConsumerRecord) {
+    var kafkaHeaders = kafkaConsumerRecord.headers();
+    var okapiConnectionParams = new OkapiConnectionParams(KafkaHeaderUtils.kafkaHeadersToMap(kafkaHeaders), vertx);
+    var event = new JsonObject(kafkaConsumerRecord.value()).mapTo(OrderAuditEvent.class);
     LOGGER.info("handle:: Starting processing of Order audit event with id: {} for order id: {}", event.getId(), event.getOrderId());
 
-    orderAuditEventsService.saveOrderAuditEvent(event, okapiConnectionParams.getTenantId())
-      .onSuccess(ar -> {
-        LOGGER.info("handle:: Order audit event with id: {} has been processed for order id: {}", event.getId(), event.getOrderId());
-        result.complete(event.getId());
-      })
+    return orderAuditEventsService.saveOrderAuditEvent(event, okapiConnectionParams.getTenantId())
+      .onSuccess(ar -> LOGGER.info("handle:: Order audit event with id: {} has been processed for order id: {}", event.getId(), event.getOrderId()))
       .onFailure(e -> {
         if (e instanceof DuplicateEventException) {
           LOGGER.info("handle:: Duplicate Order audit event with id: {} for order id: {} received, skipped processing", event.getId(), event.getOrderId());
-          result.complete(event.getId());
```

Comment:
seems we still need to use promise here, because in case of DuplicateEventException we should not return failed future.
Duplicates in kafka are possible and for us it is an expected situation and we should return successful response